tests/test_custom_learner.py

```diff
@@ -229,3 +229,23 @@ def test_custom_learner_api_missing_file():
 def test_custom_learner_api_bad_extension():
     other_dir = join(_my_dir, 'other')
     _ = Learner('_CustomLogisticRegressionWrapper', custom_learner_path=join(other_dir, 'custom_learner.txt'))
+
+
+@raises(ValueError)
+def test_custom_learner_learning_curve_min_examples():
+    """
+    test to check that Learner.learning_curve raises an error
+    when the number of training examples provided is less than 500
+    :return:
+    """
+    # generates a training split with less than 500 examples
+    train_fs_less_than_500, _ = make_classification_data(num_examples=499,
+                                                         train_test_ratio=1.0,
+                                                         num_labels=3)
+
+    # creating an example learner
+    learner = Learner('LogisticRegression')
+
+    # this must throw an error because `examples` has less than 500 items
+    _ = learner.learning_curve(examples=train_fs_less_than_500, metric='accuracy',
+                               override_minimum=False)
```

Comment:
I think we don't want to specify `override_minimum` here since the goal is to check that the error is raised by default. 

I think we need another tests, where we specify `override_minimum` as `True`, and then check that the warning was output to the log file. I believe we have other tests that already  check for warnings in the log so you should be able to adapt those.

Reply:
That makes sense - I've updated this test and have also added a separate test to check that the warning is raised: https://github.com/EducationalTestingService/skll/pull/631/files#diff-d8e9ffeba7d07b6f722f054d5a627f2582ba38b8b90ea090fb38d30cea565936R261-R292